src/Plugin/WebformElement/NetsEasyPaymentElement.php

```diff
@@ -115,30 +116,113 @@ public function alterForm(array &$element, array &$form, FormStateInterface $for
         $form['os2forms_payment_content']['#markup'] = $element['#checkout_page_description'];
       }
 
-      $form['checkout_container'] = [
+      $form['os2forms_payment_checkout_container'] = [
         '#type' => 'container',
         '#attributes' => [
           'id' => 'checkout-container-div',
           'data-checkout-key' => $this->paymentHelper->getCheckoutKey(),
-          'data-payment-error-message' => $this->t('Der skete en fejl. Prøv venligst igen senere'),
+          'data-payment-error-message' => $this->t('An error has occurred. Please try again later.'),
           'data-create-payment-url' => Url::fromRoute("os2forms_payment.createPayment",
             [
               'amountToPay' => $amount_to_pay,
               'callbackUrl' => $callback_url,
             ])->toString(TRUE)->getGeneratedUrl(),
         ],
+        '#limit_validation_errors' => [],
+        '#element_validate' => [[get_class($this), 'validateOptions']],
       ];
-      $form['payment_reference_field'] = [
+      $form['os2forms_payment_payment_reference_field'] = [
         '#type' => 'hidden',
         '#name' => 'payment_reference_field',
       ];
     }
   }
 
   /**
-   * Get recipient elements.
+   * Validates the payment object.
    *
-   * @phpstan-return array<string, mixed>
+   * @param array<mixed> $element
+   *   Form element object.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   Form state object.
+   *
+   * @return mixed
+   *   Returns validation results.
+   */
+  public static function validateOptions(array &$element, FormStateInterface $form_state): mixed {
+    $paymentHelper = \Drupal::service('Drupal\os2forms_payment\Helper\PaymentHelper');
+    $netsEasyController = \Drupal::service('Drupal\os2forms_payment\Controller\NetsEasyController');
+
+    $paymentId = $form_state->getValue('os2forms_payment_payment_reference_field');
+    if (!$paymentId) {
+      return $form_state->setError(
+        $element,
+        t('The form could not be submitted. Please try again.')
+      );
+    }
+
+    $endpoint = $paymentHelper->getTestMode()
+      ? 'https://test.api.dibspayment.eu/v1/payments/' . $paymentId
+      : 'https://api.dibspayment.eu/v1/payments/' . $paymentId;
```

Comment:
Most of this code is duplicated. Add a helper method to the payment helper to get the URL, e.g.

```php
class PaymentHelper {
  public function getPaymentUrl(string $path) {
    $baseUrl = $this->getTestMode()
      ? 'https://test.api.dibspayment.eu/'
      : 'https://api.dibspayment.eu/';

    return $baseUrl.$path;
  }

  …
}
```

and use it whenever you need a payment URL.